lib/screens/chat_page.dart

```diff
@@ -0,0 +1,195 @@
+import 'dart:convert';
+import 'package:dart_jsonwebtoken/dart_jsonwebtoken.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:prototype_1/styles/colors.dart';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+
+class ChatPage extends StatefulWidget {
+  const ChatPage({Key? key}) : super(key: key);
+
+  @override
+  State<ChatPage> createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  List<List<dynamic>> messages = [[
+    'Bonjour, je suis Edgar, votre assistant médical. Comment puis-je vous aider ?',
+    false,
+  ]];
+
+  List<String> symptomsType = ["maux_de_tetes", "vision_trouble", "fievre", "maux_de_ventre", "vomissements"];
+
+  Map<String, dynamic> symptomsContext = {
+    "maux_de_tetes": { "symptoms": "Maux de tête", "present": null },
+    "vision_trouble": { "symptoms": "Vision trouble", "present": null },
+    "fievre": { "symptoms": "Fièvre", "present": null },
+    "maux_de_ventre": { "symptoms": "Maux de ventre", "present": null },
+    "vomissements": { "symptoms": "Vomissements", "present": null },
+  };
+
+  Future<void> parseUserInput(String userInput) async {
+    await dotenv.load();
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    final token = prefs.getString('token');
+    final uncodeToken = JWT.decode(token!);
+    final payload = uncodeToken.payload;
+    // ignore: unused_local_variable
+    final id = payload['patient']['id'];
+    final url = '${dotenv.env['URL']}/nlp';
+    final urlexam = '${dotenv.env['URL']}/exam';
+    final responseNLP = await http.post(
+    Uri.parse(url),
+      headers: {
+      'Content-Type': 'application/json',
+      },
+      body: jsonEncode({
+      'symptoms': symptomsType, // Replace 'symptomsType' with the actual list of symptoms
+      'input': userInput,
+      }),
+    );
+
+    if (responseNLP.statusCode == 200) {
+      final nlpData = jsonDecode(responseNLP.body);
+      final context = nlpData['context']; 
+
+      final responseExam = await http.post(
+      Uri.parse(urlexam),
+      headers: {
+      'Content-Type': 'application/json',
+      },
+      body: jsonEncode({
+      'context': context,
+      }),
+      );
+
+      if (responseExam.statusCode == 200) {
+      final examData = jsonDecode(responseExam.body);
+      final nextQuestion = examData['question'];
+      // ignore: unused_local_variable
+      final symptoms = examData['symptoms'];
+      final done = examData['done'];
+
+      if (done) {
+        setState(() {
+        messages.add([nextQuestion, false]);
+        });
+      } else {
+        setState(() {
+          messages.add([nextQuestion, false]);
+          });
+        }
+      } else {
+        setState(() {
+        messages.add(['Une erreur est survenue lors de la requête d\'examen', false]);
+        });
+      }
```

Comment:
Be careful of your line indentation, it's too complicated to read the last lines. I thought that you added two time `else` for the same `if` but no it's just your indentation